File: app/src/main/java/com/amarnehsoft/holyquran/model/Aya.java
package com.amarnehsoft.holyquran.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

import java.util.HashMap;
import java.util.Map;

@Entity
public class Aya{
    @PrimaryKey
    private int number;

    private String text;
    private int numberInSurah;
    private int juz;
    private int manzil;
    private int page;
    private int ruku;
    private int hizbQuarter;
    private String sajda;

    @Ignore
    private Surah surah;

    private int surahNumber;
    private String audio;

    private String tafseer;
    private String translation;

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public int getNumberInSurah() {
        return numberInSurah;
    }

    public void setNumberInSurah(int numberInSurah) {
        this.numberInSurah = numberInSurah;
    }

    public int getJuz() {
        return juz;
    }

    public void setJuz(int juz) {
        this.juz = juz;
    }

    public int getManzil() {
        return manzil;
    }

    public void setManzil(int manzil) {
        this.manzil = manzil;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getRuku() {
        return ruku;
    }

    public void setRuku(int ruku) {
        this.ruku = ruku;
    }

    public int getHizbQuarter() {
        return hizbQuarter;
    }

    public void setHizbQuarter(int hizbQuarter) {
        this.hizbQuarter = hizbQuarter;
    }

    public String getSajda() {
        return sajda;
    }

    public void setSajda(String sajda) {
        this.sajda = sajda;
    }

    public Surah getSurah() {
        return surah;
    }

    public void setSurah(Surah surah) {
        this.surah = surah;
    }

    public String getAudio() {
        return audio;
    }

    public void setAudio(String audio) {
        this.audio = audio;
    }


    public String getTafseer() {
        return tafseer;
    }

    public void setTafseer(String tafseer) {
        this.tafseer = tafseer;
    }

    public String getTranslation() {
        return translation;
    }

    public void setTranslation(String translation) {
        this.translation = translation;
    }

    public int getSurahNumber() {
        return surahNumber;
    }

    public void setSurahNumber(int surahNumber) {
        this.surahNumber = surahNumber;
    }
}

File: app/src/main/java/com/amarnehsoft/holyquran/model/Aya.kt
package com.amarnehsoft.holyquran.model

import com.amarnehsoft.holyquran.network.quran.Quran

data class Aya(
    var number: Int,
    var text: String,
    var numberInSurah: Int,
    var juz: Int,
    var manzil: Int,
    var page: Int,
    var ruku: Int,
    var hizbQuarter: Int
//    var surahNumber: Int
    //var audio: String,
    //var tafseer: String = "",
    //var translation: String = "",
    //var sajda: Any? = null,
//    var surah: Surah
) {
    companion object {
        fun create(apiAya: Quran.Ayah): Aya {
            return Aya(
                number = apiAya.number,
                text = apiAya.text,
                numberInSurah = apiAya.numberInSurah,
                juz = apiAya.juz,
                manzil = apiAya.manzil,
                page = apiAya.page,
                ruku = apiAya.ruku,
                hizbQuarter = apiAya.hizbQuarter
            )
        }
    }
}
